engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -134,10 +135,19 @@ class WorkflowManagerActor(params: WorkflowManagerActorParams)
   private val maxWorkflowsToLaunch = config.getConfig("system").as[Option[Int]]("max-workflow-launch-count").getOrElse(DefaultMaxWorkflowsToLaunch)
   private val newWorkflowPollRate = config.getConfig("system").as[Option[Int]]("new-workflow-poll-rate").getOrElse(DefaultNewWorkflowPollRate).seconds
   private val fileHashCacheEnabled = config.as[Option[Boolean]]("system.file-hash-cache").getOrElse(false)
+  private val callCachingConfig = config.getConfig("call-caching")
 
+  private val errorOrCallCachingEnabled = errorOrCallCachingBoolean("enabled")
```

Comment:
In My Humble Opinion: this (and the `val errorOrInvalidateBadCacheResults` below) should fail-fast and crash **here** on an invalid `Boolean` string, not wrap an invalid config then pass the `ErrorOr[Boolean]` around. That would be similar and consistent with the above lines such as:
```scala
private val fileHashCacheEnabled = config.as[Option[Boolean]]("system.file-hash-cache").getOrElse(false)
```

That line will throw a parsing exception **on that line** for bad `Boolean` strings like `tru` or `fales`.

As both config keys have default values defined in the [reference.conf](https://github.com/broadinstitute/cromwell/blob/45/core/src/main/resources/reference.conf#L259-L264), one could just do:
```scala
private val callCachingEnabled = callCachingConfig.getBoolean("enabled")
private val invalidateBadCacheResults = callCachingConfig.getBoolean("invalidate-bad-cache-results")
```

That will then crash cromwell on startup with an invalid config, satisfying [BA-5923](https://broadworkbench.atlassian.net/browse/BA-5923), and make the parameter passing a bit easier using `Boolean` instead of `ErrorOr[Boolean]`.

There is a pre-existing "bad UX" issue with the config parsing in this area of code though. Say someone accidentally configures Cromwell like this:

```hocon
call-caching.enabled = tru
call-caching.invalidate-bad-cache-results = fales
```

Based on everything discussed above, only the first invalid config key `call-caching.enabled` will throw an error on startup.

Alternatively, if one wanted this code be truly "[Validated](https://typelevel.org/cats/datatypes/validated.html)", one could convert **all** of the config parsers in this section to `ErrorOr`, and then do a single `.mapN()` to return all the errors at once. That processing could still live in this area of code such that the `WorkflowManagerActor` doesn't start if the config file is invalid.

That said, as long as the behavior is consistent (config parsing fails fast - or - aggregates all `ErrorOr[_]`, but not both) then I think it's fine.

Reply:
Thank you for your review and suggestions! PR is updated.